### DISCUSSION TITLE
Disable requiring addons possible?

### DISCUSSION THREAD

**Discussion by techar** [nodejs/help#4901] [Q&A]
I'm using `vm2 `sandbox to only allow requiring some internal modules but how do you block requiring the Node Addons? They might as well be doing the same things like the denied core modules, like spawning processes, reading restricted files etc. 

My vm2 config is:
  ```
new NodeVM({
    require: {
        external: true,
        builtin: [x,y,z],
        mock: {..},
        context: 'sandbox'
    }..
```

**Comment by cjihrig** [comment]
It looks like this question was specifically about the `vm2` npm module (and was answered [here](https://github.com/patriksimek/vm2/issues/327)), and not related to Node core.

If you want to do this in Node core, you can use the [`--no-addons`](https://nodejs.org/api/cli.html#--no-addons) CLI flag.
